tf2/src/buffer_core.cpp

```diff
@@ -528,20 +529,22 @@ tf2::TF2Error BufferCore::walkToTopParent(F& f, TimePoint time, CompactFrameID t
   if (frame_chain)
   {
     // Pruning: Compare the chains starting at the parent (end) until they differ
-    int m = (int)reverse_frame_chain.size()-1;
-    int n = (int)frame_chain->size()-1;
-    for (; m >= 0 && n >= 0; --m, --n)
+    size_t m = reverse_frame_chain.size()-1;
+    size_t n = frame_chain->size()-1;
+    for (; m != std::numeric_limits<size_t>::max() &&
+      n != std::numeric_limits<size_t>::max(); --m, --n)
```

Comment:
@ivanpauno I get the point of using unsigned integers, but conditions become unusual. Why not reversing the indexes to count up?

Reply:
Yes, I wanted to preserve the original logic, but reversing the indexing sounds reasonable

Reply:
Done in https://github.com/ros2/geometry2/pull/270/commits/2918e19733abed5fd357d0fa0902653502752d7b.

Please, carefully check the logic is equivalent, as I'm not sure at what extent this is being tested.